File: c_src/28.c
#include "../include/my.h"


int check(char* s1, char* s2, int len)
{
    for (int i = 0; i < len; i++) {
        if (s1[i] != s2[i]) {
            return 0;
        }
    }
    return 1;
}

int myStrStr(char * haystack, char * needle)
{
    int len1 = strlen(haystack);
    int len2 = strlen(needle);
    if (len2 == 0) {
        return 0;
    }
    if (len1 < len2) {
        return -1;
    }
    for (int i = 0; i <= len1 - len2; i++) {
        if (check(haystack + i, needle, len2)) {
            return i;
        }
    }
    return -1;
}

File: go_src/28.go
func strStr(haystack string, needle string) int {
	len1, len2 := len(haystack), len(needle)
	if len1 < len2 {
		return -1
	}
	if len2 == 0 {
		return 0
	}
	index := 0
	for index <= len1 - len2 {
		curIndex := 0
		for ; curIndex < len2; curIndex++ {
			if haystack[index + curIndex] != needle[curIndex] {
				break
			}
		}
		if curIndex == len2 {
			return index
		}
		index++
	}
	return -1
}
